packages/react-client/src/hooks/connection.ts

```diff
@@ -1,36 +1,24 @@
-import { useFishjamContext } from "./fishjamContext";
-import type { FishjamContextType, UseConnect } from "../types";
+import type { UseConnect } from "../types";
 import { useCallback, useMemo } from "react";
-import type { UseReconnection } from "../state.types";
+import { useFishjamClient } from "./tsClient";
 
 export function useConnect(): UseConnect {
-  const { state }: FishjamContextType = useFishjamContext();
+  const client = useFishjamClient();
 
   return useMemo(() => {
     return (config) => {
-      state.client.connect(config);
+      client.connect(config);
       return () => {
-        state.client.disconnect();
+        client.disconnect();
```

Comment:
**Discussion Idea**: If we have `useDisconnect`, should we return anything from the `connect()` function? Maybe we could return a promise that resolves when the user establishes a connection.

Reply:
big up, i see no benefit from returning the disconnect fn, then some benefit of making the connect awaitable

Reply:
https://linear.app/swmansion/issue/FCE-460/return-promise-from-connect-function